### PR TITLE
Add parameter for environment variables to be set before analyzing dscript so that they can be fed from the output of another pip

### PR DESCRIPTION
**Pull request by erickulcyk**

### CODE REVIEW

File: Public/Src/Tools/UnitTests/BxlScriptAnalyzer/AnalyzerTest.cs [hunk 1]
```diff
@@ -121,10 +122,10 @@ public class AnalyzerTest<TAnalyzer> : TemporaryStorageTestBase
             bool preserveTrivia = false)
         {
             var analyzer = new TAnalyzer();
-
+            var pathTable = new PathTable();
             var args = new Args(
-                config: @"b:\Fake.config.dsc",
-                filter: string.Empty,
+                commandLineConfig: new CommandLineConfiguration() { Startup = { ConfigFile = AbsolutePath.Create(pathTable, @"b:\Fake.config.dsc") } },
```
Comment (almiliMSFT) [inline Public/Src/Tools/UnitTests/BxlScriptAnalyzer/AnalyzerTest.cs]:
```suggestion
                commandLineConfig: new CommandLineConfiguration() { Startup = { ConfigFile = AbsolutePath.Create(pathTable, X("b/Fake.config.dsc")) } },
```

Comment (almiliMSFT) [reply Public/Src/Tools/UnitTests/BxlScriptAnalyzer/AnalyzerTest.cs]: this should fix the Mac test failures

**Comment by erickulcyk** [comment]
azp run public builds pr validation

**Pull request merged**